### PR TITLE
IMyShipConnector: ConnectStatus added, and comments

### PR DESCRIPTION
**Pull request by Sibz**
Added read only tri-state property ConnectStatus
Added comments to all properties.
Added Obsolete flag to IsLocked and IsConnected

### CODE REVIEW

**Comment by lord-devious** [comment]
Isn't this what IsLocked does?

**Comment by mrsib** [comment]
Not Quite, IsLocked returns true when ready to lock and when connected. where as this returns true only when ready to lock.
To be honest the naming of IsLocked is confusing and perhaps should be deprecated.

**Comment by lord-devious** [comment]
Oh, I see! I think I like that. And I agree about IsLocked, that's why I added IsConnected (and OtherConnector). Actually your property will deprecate IsLocked, I suggest you add an Obsolete tag to it.

**Comment by mexmer** [comment]
would not be better to make it tristate and obsolete both of them?

**Comment by mrsib** [comment]
@mexmer whats the best way to do that? I understand Enums don't work for in game, so would it be best to return string. i.e. "OutOfRange","ReadyToLock", "Connected"?

**Comment by lord-devious** [comment]
Enums created in game doesn't work, API created ones does. And I'll tend to agree with @mexmer , it's what I did for doors and pistons. Question is how many obsolete properties we want. At some point one just has to accept how things are.

**Comment by mrsib** [comment]
@mexmer @lord-devious 
Well funny, my original intention was for a tri-state property, but wasn't sure on enums and saw IsConnected was already there so decided to just add IsReadyToLock

Added the above commit, Is the ConnectorState Enum structure good enough? Is it in the right place?

**Comment by lord-devious** [comment]
Heh that's what went through my mind when I added IsConnected :) (although I knew enums work) 
I don't really like obsoleting stuff, because it leaves a mess in the files so I went through that 

Looks good to me although it should have its own file.

**Comment by Sibz** [comment]
@lord-devious Yeah own file probably best, but I wouldn't know where to put it in the giant folder structure

**Comment by Sibz** [comment]
@lord-devious I see what you did with pistonstatus, I'll do that.

**Comment by Sibz** [comment]
@lord-devious done, and renamed to status to keep naming in line with pistonstatus

File: Sources/Sandbox.Common/ModAPI/Ingame/IMyShipConnector.cs [hunk 1]
```diff
@@ -7,10 +7,43 @@ namespace Sandbox.ModAPI.Ingame
...
+        /// <summary>
+        /// Connector is set to pull items from connected inventories (Read Only)
+        /// </summary>
         bool CollectAll { get; }
+
+        /// <summary>
+        /// DEPRECATED use ConnectState
+        /// Connector is working and in range of another connector, could be connected or ready to lock (Read Only)
+        /// </summary>
+        [Obsolete("Deprecated, Use ConnectState instead")]
```
Comment (harag-on-steam) [inline Sources/Sandbox.Common/ModAPI/Ingame/IMyShipConnector.cs]: There is no ConnectState any more ;-)

Comment (Sibz) [inline Sources/Sandbox.Common/ModAPI/Ingame/IMyShipConnector.cs]: Haha, Thanks, Fixed.

**Comment by mexmer** [comment]
@lord-devious it doesn't need to be in separate file, if it's used only by that interface, it's better if enum is same file as interface

**Comment by lord-devious** [comment]
I disagree. All types should be in their own files, much cleaner that way. Anyway it's up to OP. I was merely stating my opinion.

**Comment by Sibz** [comment]
I am thinking both are right, its nice to have it cleanly in its own file, however lots of little enums will clutter the file list and it can be in same file as interface.
As PistonStatus is separate, and the Devs oversaw the introduction of that, I will keep it the same. It can be tidied up later if necessary.

**Comment by Sibz** [comment]
Conflicts

**Pull request closed without merging**